migrations/coin_policy_migration.go

```diff
@@ -0,0 +1,88 @@
+package migrations
+
+import "C"
+import (
+	loom "github.com/loomnetwork/go-loom"
+	ctypes "github.com/loomnetwork/go-loom/builtin/types/coin"
+	"github.com/loomnetwork/go-loom/plugin/contractpb"
+	"github.com/loomnetwork/go-loom/types"
+	"github.com/loomnetwork/loomchain"
+	"github.com/loomnetwork/loomchain/config"
+	"github.com/pkg/errors"
+)
+
+var (
+	InvalidBaseMintingAmount = errors.New("Base Minting Amount should be greater than zero")
+)
+
+type (
+	Policy = ctypes.Policy
+)
+
+var (
+	policyKey = []byte("policy")
+)
+
+func GenerateCoinPolicyMigrationFn(cfg *config.Config) func(ctx *MigrationContext) error {
+	return func(ctx *MigrationContext) error {
+		//Resolve coin context
+		_, coinCtx, err := resolveCoin(ctx)
+		if err != nil {
+			return err
+		}
+		div := loom.NewBigUIntFromInt(10)
+		div.Exp(div, loom.NewBigUIntFromInt(18), nil)
+		if len(cfg.CoinPolicyMigrationConfig.MintingAccount) == 0 {
```

Comment:
Also, check if `cfg.CoinPolicyMigrationConfig` is nil

Reply:
Done.

Reply:
@suhasagg hmm. This validation code has become too long. Let's do one thing, move it as a method for `CoinPolicyMigrationConfig` as in 

```
func (c *CoinPolicyMigrationConfig) IsValid() error {
   // Validate all fields here
}
```

and in here, we only validate if `cfg.CoinPolicyMigrationConfig` is nil or not.